### PR TITLE
Added sparse items array support

### PR DESCRIPTION
**Pull request by voy**

### CODE REVIEW

**Comment by voy** [comment]
Don't merge at this time.

File: src/InfiniteList.jsx [hunk 1]
```diff
@@ -111,42 +114,59 @@ export default class InfiniteList extends React.Component {
...
+    }
 
-        var visibleItems = items.slice(renderedStart, renderedStart + numOfVisibleItems);
-        var listItems = visibleItems.map(this._getItemComponent, this);
+    _getContentStyle(itemsPerPage) {
+        var { itemHeight } = this.props;
 
-        const dataRangeEnd = Math.min(renderedStart + listItems.length, this.props.items.length);
-        this.props.paging && this._notifyWhenDataIsNeeded(renderedStart, dataRangeEnd);
+        // the number one guarantees there is never empty space at the end of the list
```
Comment (vaclavbohac) [inline src/InfiniteList.jsx]: Is this still relevant?

File: src/InfiniteList.jsx [hunk 2]
```diff
@@ -111,42 +114,59 @@ export default class InfiniteList extends React.Component {
...
     }
 
-    _notifyWhenDataIsNeeded(start, end) {
-        const items = this.props.items;
+    _getVisibleSlice(items, start, end) {
+        let result = [];
 
-        // Do not go over the end of the array
-        if (end >= items.length ) end = items.length - 1;
+        for (let i = start; i < end; i++) {
```
Comment (vaclavbohac) [inline src/InfiniteList.jsx]: Seems like `_.slice` to me.

Comment (vaclavbohac) [inline src/InfiniteList.jsx]: But it seems you need to keep indexes. Ouch

**Pull request merged**